content/en/docs/tutorials/configuration/configure-java-microservice/configure-java-microservice.md

```diff
@@ -7,62 +7,235 @@ weight: 10
+
+### ConfigMaps: Storing Non-Sensitive Data
+
+ConfigMaps are perfect for storing non-sensitive configuration data that can be shared across multiple containers. They are Kubernetes API Objects that store non-confidential key-value pairs.
+
+**Step 1: Create a ConfigMap**
+
+To get started, let's create a ConfigMap to store your microservice's application name. Follow these steps:
+
+1. Open a text editor of your choice.
+
+2. Create a new file with a `.yaml` extension, for example, `myapp-configmap.yaml`.
+
+3. Copy and paste the following content into the `myapp-configmap.yaml` file:
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: myapp-config
+      annotations:
+        description: "ConfigMap to store application name and API endpoint"
+      labels:
+        app: myapp
+        environment: production
+      namespace: my-namespace
+    data:
+      app.name: MyApp
+      api.endpoint: https://api.example.com
+    ```  
+
+  In this YAML definition:
+   - `apiVersion` specifies the Kubernetes API version to use for the resource.
+   - `kind` indicates the type of resource, which in this case is a ConfigMap.
+   - `metadata` contains metadata about the ConfigMap, including its name, annotations, labels, and namespace.
+   - `data` is where you define the key-value pairs that make up the content of the ConfigMap.
+
+   The `annotations` section is used to attach non-identifying metadata to the resource. Labels are key-value pairs used for identifying and grouping resources, and the `namespace` field specifies the Kubernetes namespace where the ConfigMap will be created.
+
+Here, we're creating a ConfigMap named `myapp-config` with a key-value pair: `app.name: MyApp`. This ConfigMap will serve as a centralized storage for your microservice's configuration data, making it easily accessible to your application. For more information regarding ConfigMaps, you can refer to official documentation [here](/docs/tasks/configure-pod-container/configure-pod-configmap/).
+
+### Secrets: Safeguarding Sensitive Information
+
+Secrets are designed to securely manage confidential data like passwords and API keys.
+
+**Step 2: Create a Secret**
+
+In this step, you'll create a Kubernetes Secret to securely store sensitive data, such as passwords and API keys. You'll define the Secret using YAML format and ensure that sensitive information is base64-encoded for added security.
+
+Here's how you can proceed:
+
+1. Open a text editor of your choice.
+
+2. Create a new file with a `.yaml` extension, for example, `myapp-secret.yaml`.
+
+3. Copy and paste the following content into the `myapp-secret.yaml` file:
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: myapp-secret
+    type: Opaque
+    data:
+      db.password: <base64-encoded-password>
+      api.key: <base64-encoded-api-key>
+    ```
```

Comment:
If possible, use a `code` shortcode: https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files. It's OK to encourage the reader to type it in themselves, but let them choose.